cmake/modules/RootMacros.cmake

```diff
@@ -836,49 +960,47 @@ function(ROOT_LINKER_LIBRARY library)
 endfunction()
 
 #---------------------------------------------------------------------------------------------------
-#---ROOT_ADD_INCLUDE_DIRECTORIES( library )
+#---ROOT_ADD_INCLUDE_DIRECTORIES( library DEPENDENCIES dep1 dep2 ...)
```

Comment:
I will check if cmake really propogate include paths for dependend libraries - I am not sure if it works this way.
 
Of course, I useing `target_include_directories` here

Reply:
@Axel-Naumann I have to specify `DEPENDENCIES` here. target does not automatically inherit `includes_directories` from dependent libs - at least not here

Reply:
...which is why we should be using `target_include_directories`.

Reply:
I using `target_include_directories` everywhere but I still need `DEPENDENCIES` to extract all includes I need to specify for target.

Reply:
Then we are not declaring these dependencies at the caller correctly. Let's look at one example. Actually, is this the only case where we need it? https://github.com/root-project/root/pull/5181/files#diff-7d50d20c138eee72be054eab0da41324R49-R50

So - why do we need `DEPENDENCIES` instead of passing `GLEW` to `LIBRARIES`? @oshadura do I guess correctly that this is gone anyway after your glew PR? Then we really really should not introduce this new parameter `DEPENDENCIES`!

Reply:
It has nothing to with GLEW. 

Simple example. If one uses `TFile.h` in the library, `RIO` must be in list of library dependencies. And `DEPENDENCIES` parameter is a way to extract information, that `${CMAKE_SOURCE_DIR/io/io/inc` should be in include path for such library. 

And such include is not automatically extracted from library dependencies. We already using `PUBLIC` specifier in `target_include_directories`.

Reply:
Let's postpone to a voice discussion (tomorrow morning after team round table?)